db/data_migration/20230424100641_add_initial_worldwide_organisation_audits.rb

```diff
@@ -0,0 +1,19 @@
+EVENT_TYPE = "initial".freeze
```

Comment:
Couple of thoughts about this

1. Is this the most sensible thing to do? We could use the `updated_at` to set the `created_at` of the `Version` instead.
2. Am I correct that there are only 435 worldwide organisations, so we don't need to worry about batching the requests.
3. How do we ensure that we run this migration before any users make further changes to worldwide organsiations (that will now be audited). Could we move this to the normal migrations folder so it runs automatically, or is there another approach that we take on this app?

Reply:
> How do we ensure that we run this migration before any users make further changes to worldwide organsiations (that will now be audited). Could we move this to the normal migrations folder so it runs automatically, or is there another approach that we take on this app?

Good question. Unfortunately even if this was moved to the normal migrations folder, there's still the risk that users will make changes before it's run – the hosting platform [makes no guarantee](https://govuk-k8s-user-docs.publishing.service.gov.uk/manage-app/access-ci-cd/#deploying-a-release-of-a-gov-uk-app) that database migrations will complete before new code hits the database.

I think in practice, this is very unlikely to be a problem. However it's probably best not to tempt fate.

I think the only foolproof way to achieve this is to split it into a two-stage deployment:

1. Deploy & run the "create initial audits" data migration first
2. Then follow up with the other changes, including where `include AuditTrail` is introduced to the `WorldwideOrganisation` model

Reply:
@jkempster34

> In terms of design, how do you feel about "No history before this time" ..."user (unknown)". James suggested another approach https://github.com/alphagov/whitehall/pull/7583#discussion_r1177925906 that's worth considering.

I think the current approach you've taken seems reasonable – i.e. a "No history before this time" event with today's date (or the date this gets deployed/migrated). I think that's the most truthful representation of the situation. It'll be the date we started tracking edit history.

Whereas it feels a bit misleading to backdate 'created' and 'updated' events based on the timestamps in the records – since it'd give an incomplete picture of the edit history.